File: packages/react-native-client/android/src/main/java/io/fishjam/reactnative/managers/TracksUpdateListenersManager.kt
package io.fishjam.reactnative.managers

interface TrackUpdateListener {
  fun onTracksUpdate()
}

class TracksUpdateListenersManager : ListenerManager<TrackUpdateListener>() {
  fun notifyListeners() {
    for (listener in listeners) listener.onTracksUpdate()
  }
}

File: packages/react-native-client/ios/managers/TracksUpdateListenersManager.swift
protocol TrackUpdateListener: AnyObject {
    func onTracksUpdate()
}

class TracksUpdateListenersManager {
    private var listeners: [TrackUpdateListener] = []

    func add(_ listener: TrackUpdateListener) {
        listeners.append(listener)
    }

    func remove(_ listener: TrackUpdateListener) {
        listeners.removeAll { $0 === listener }
    }

    func notifyListeners() {
        for listener in listeners {
            listener.onTracksUpdate()
        }
    }
}
